rclcpp/test/test_node.cpp

```diff
@@ -260,3 +262,1609 @@ TEST_F(TestNode, now) {
+
+    auto results = node->get_parameter_types({name1, name2});
+
+    EXPECT_EQ(results.size(), 2u);
+
+    EXPECT_EQ(results[0], rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER);
+
+    EXPECT_EQ(results[1], rcl_interfaces::msg::ParameterType::PARAMETER_STRING);
+  }
+  {
+    // non-existent parameter throws
+    auto name = "parameter"_unq;
+
+    {
+      EXPECT_THROW({
+        node->get_parameter_types({name});
+      }, rclcpp::exceptions::ParameterNotDeclaredException);
+    }
+  }
+  {
+    // check that repeated names in input work, and that output is stable (same order as input)
+    auto name1 = "parameter"_unq;
+    auto name2 = "parameter"_unq;
+
+    node->declare_parameter(name1, 42);
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.read_only = true;
+    node->declare_parameter<std::string>(name2, "test", descriptor);
+
+    auto results = node->get_parameter_types({name2, name1, name2});
+
+    EXPECT_EQ(results.size(), 3u);
+
+    EXPECT_EQ(results[0], rcl_interfaces::msg::ParameterType::PARAMETER_STRING);
+
+    EXPECT_EQ(results[1], rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER);
+
+    EXPECT_EQ(results[2], rcl_interfaces::msg::ParameterType::PARAMETER_STRING);
+  }
+}
+
+// test get parameter types with undeclared allowed
+TEST_F(TestNode, get_parameter_types_undeclared_parameters_allowed) {
+  auto node = std::make_shared<rclcpp::Node>(
+    "test_get_parameter_types_node"_unq,
+    rclcpp::NodeOptions().allow_undeclared_parameters(true));
+  {
+    // normal use still works (declare first)
+    auto name1 = "parameter"_unq;
+    auto name2 = "parameter"_unq;
+
+    node->declare_parameter(name1, 42);
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.read_only = true;
+    node->declare_parameter<std::string>(name2, "test", descriptor);
+
+    auto results = node->get_parameter_types({name1, name2});
+
+    EXPECT_EQ(results.size(), 2u);
+
+    EXPECT_EQ(results[0], rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER);
+
+    EXPECT_EQ(results[1], rcl_interfaces::msg::ParameterType::PARAMETER_STRING);
+  }
+  {
+    // non-existent parameter does not throw
+    auto name = "parameter"_unq;
+
+    auto results = node->get_parameter_types({name});
+
+    EXPECT_EQ(results.size(), 1u);
+
+    EXPECT_EQ(results[0], rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET);
+  }
+  {
+    // check that repeated names in input work, and that output is stable (same order as input)
+    auto name1 = "parameter"_unq;
+    auto name2 = "parameter"_unq;
+
+    auto results = node->get_parameter_types({name2, name1, name2});
+
+    EXPECT_EQ(results.size(), 3u);
+
+    EXPECT_EQ(results[0], rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET);
+
+    EXPECT_EQ(results[1], rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET);
+
+    EXPECT_EQ(results[2], rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET);
+  }
+}
+
+// // test get_parameter with undeclared allowed
```

Comment:
I think this commented section can be removed? Looks like they're duplicates of tests above.

Reply:
Oops didn't mean to commit that, I'll remove them.